feat: add olive support (#6)
eduNEXT/tutor-contrib-mfe-extensions: README.md

| Open edX release | Tutor version     | Tutor MFE Version                    | Plugin release |
|------------------|-------------------|--------------------------------------|----------------|

> | Nutmeg           | `>=13.0, <14`     | `edunext/tutor-mfe@14.0.2.post1`     | 14.x.x          |
> | Olive           | `>=14.0, <15`     | `edunext/tutor-mfe@15.0.7.post2`     | 15.x.x          |

Reviewer: ```suggestion
| Nutmeg           | `>=14.0, <15`     | `edunext/tutor-mfe@14.0.2.post1`     | 14.x.x          |
| Olive           | `>=15.0, <16`     | `edunext/tutor-mfe@15.0.7.post2`     | 15.x.x          |
```

- Authn: [`eduNEXT/frontend-app-authn@ednx-release/olmo.master`](https://github.com/eduNEXT/frontend-app-authn/tree/ednx-release/olmo.master)
- Account: [`eduNEXT/frontend-app-account@ednx-release/olmo.master`](https://github.com/eduNEXT/frontend-app-account/tree/ednx-release/olmo.master)
- Discussions: [`eduNEXT/frontend-app-discussions@ednx-release/olmo.master`](https://github.com/eduNEXT/frontend-app-discussions/tree/ednx-release/olmo.master)
- Gradebook: [`eduNEXT/frontend-app-gradebook@ednx-release/olmo.master`](https://github.com/eduNEXT/frontend-app-gradebook/tree/ednx-release/olmo.master)
- Ora Grading: [`eduNEXT/frontend-app-ora-grading@ednx-release/olmo.master`](https://github.com/eduNEXT/frontend-app-ora-grading/tree/ednx-release/olmo.master)
- Communications: [`eduNEXT/frontend-app-communications@ednx-release/olmo.master`](https://github.com/eduNEXT/frontend-app-communications/tree/ednx-release/olmo.master)

In addition, we need to use a custom version of `tutor-mfe` that includes the patch
`mfe-dockerfile-production-final` introduced in [`overhangio/tutor-mfe#179`](https://github.com/overhangio/tutor-mfe/pull/179).

### Building Caveats

> The value of `publicPath` is fixed at build time, therefore it's not possible to reuse
> the MFE docker image if we set it's value to a specific CDN endpoint.
> To avoid this limitation we use a script
> [`docker-entrypoint.sh`](tutormfe_extensions/templates/mfe/build/mfe/docker-entrypoint.sh)
> to dinamically change a placeholder value used in `publicPath` each time the container is
> started.
> 
> We need to make sure our image is built using the placeholder `MFE_EXTENSIONS_PLACEHOLDER_STRING/app_name` as our `publicPath` (where `app_name` refers to the value of `MFE_MYMFE_MFE_APP.name`). In Nutmeg you configure `PUBLIC_PATH` through the `config.yml` file:
> 
> ```yaml
> MFE_LEARNING_MFE_APP:
>   env:
>     production:
>       PUBLIC_PATH: MFE_EXTENSIONS_PLACEHOLDER_STRING/learning
>       MFE_CONFIG_API_URL: /api/mfe_config/v1
>       APP_ID: learning
>   name: learning
>   port: 2000
>   repository: https://github.com/eduNEXT/frontend-app-learning.git
>   version: ednx-release/olmo.master
> ```
> 
> The learning MFE in eduNEXT fork also has support for the [`MFE_CONFIG_API`](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/mfe_config_api/docs/decisions/0001-mfe-config-api.rst) in Nutmeg, you must also include the variables `MFE_CONFIG_API_URL` and `APP_ID` in your
> `config.yml` while building the MFE to enable the API.

Reviewer: ```suggestion
The value of `publicPath` is fixed at build time, therefore it's not possible to reuse
the MFE docker image if we set it's value to a specific CDN endpoint.
To avoid this limitation we use a script
[`docker-entrypoint.sh`](tutormfe_extensions/templates/mfe/build/mfe/docker-entrypoint.sh)
to dinamically change a placeholder value used in `publicPath` each time the container is
started. The plugin relies on a few Tutor patches that aren't available in Tutor v15 to do this,
for that reason make sure you use the version specified in the Installation section.

```

> Once you enable the plugin and add the previous snippet to your configuration you can build
> an image with a _"dynamic"_ `publicPath`. By default the placeholder will be replaced by
> an empty string, which was the original value. In a Kubernetes environment it will be
> replaced with the value of `MFE_EXTENSIONS_CDN_URL`.

Reviewer: ```suggestion
Once you enable the plugin you can build an image with a _"dynamic"_ `publicPath`.
By default the placeholder will be replaced by an empty string, which was the original value.
In a Kubernetes environment it will be replaced with the value of `MFE_EXTENSIONS_CDN_URL`.
```
